mk/submodule.mk

```diff
@@ -26,3 +26,34 @@ $(LIBTUV_PATH)/include:
 
 $(LIBTUV_LIBRARY):
 	$(MAKE) -C $(LIBTUV_PATH) TUV_BUILD_TYPE=release TUV_CREATE_PIC_LIB=yes TUV_PLATFORM=$(LIBTUV_PLATFORM) TUV_BOARD=$(LIBTUV_BOARD)
+
+# librabbitmq related variables
+LIBRABBITMQ_PATH = deps/rabbitmq-c
+LIBRABBITMQ_INCLUDE := -I $(LIBRABBITMQ_PATH)/build/include
+LIBRABBITMQ_LIB_PATH := $(LIBRABBITMQ_PATH)/build/librabbitmq/
+ifeq ($(UNAME_S),darwin)
+    # macOS
+    LIBRABBITMQ_LINK := -Wl,-rpath,$(LIBRABBITMQ_LIB_PATH) -L$(LIBRABBITMQ_LIB_PATH) -lrabbitmq
+    LIBRABBITMQ_LIBRARY := $(LIBRABBITMQ_LIB_PATH)/librabbitmq.dylib
+else
+    LIBRABBITMQ_LINK := -Wl,-rpath=$(LIBRABBITMQ_LIB_PATH) -L$(LIBRABBITMQ_LIB_PATH) -lrabbitmq
+    LIBRABBITMQ_LIBRARY := $(LIBRABBITMQ_LIB_PATH)/librabbitmq.so
+endif
+
+$(LIBRABBITMQ_PATH)/build/include:
+	git submodule update --init $(LIBRABBITMQ_PATH)
+	mkdir $(LIBRABBITMQ_PATH)/build
+ifeq ($(UNAME_S),darwin)
+	# macOS
+	cd $(LIBRABBITMQ_PATH)/build && \
```

Comment:
I wonder whether this is really necessary to enter the `$(LIBRABBITMQ_PATH)/build` directory.

Reply:
I think that using `cmake`  keeps simple as entering $(LIBRABBITMQ_PATH)/build directory to build and configure RabbitMQ.

Reply:
For example, can't it be like:
```
cmake -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl/ -DCMAKE_INSTALL_PREFIX=./build . && \
cmake --build ./build --target install
```

Reply:
Error as entering $(LIBRABBITMQ_PATH) and using the above command to build as follows:
```
-- Found POPT: /usr/include
-- Found XMLTO: /usr/bin/xmlto
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Building rabbitmq as a shared library - yes
-- Building rabbitmq as a static library - yes
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ajubuntu/Desktop/dcurl/deps/rabbitmq-c
Error: could not load cache
```